pkg/framework/aries/api/vdr/vdr.go

```diff
@@ -24,17 +23,96 @@ const DIDCommServiceType = "did-communication"
 
 // Registry vdr registry.
 type Registry interface {
-	Resolve(did string, opts ...resolve.Option) (*did.DocResolution, error)
+	Resolve(did string, opts ...ResolveOption) (*did.DocResolution, error)
 	Store(doc *did.Doc) error
-	Create(method string, opts ...create.Option) (*did.DocResolution, error)
+	Create(method string, did *did.Doc, opts ...CreateOption) (*did.DocResolution, error)
 	Close() error
 }
 
 // VDR verifiable data registry interface.
 type VDR interface {
-	Read(did string, opts ...resolve.Option) (*did.DocResolution, error)
-	Store(doc *did.Doc, by *[]doc.ModifiedBy) error
-	Build(keyManager kms.KeyManager, opts ...create.Option) (*did.DocResolution, error)
+	Read(did string, opts ...ResolveOption) (*did.DocResolution, error)
+	Store(doc *did.Doc, by *[]ModifiedBy) error
+	Build(keyManager kms.KeyManager, did *did.Doc, opts ...CreateOption) (*did.DocResolution, error)
 	Accept(method string) bool
 	Close() error
 }
+
+// ModifiedBy key/signature used to update the DID Document.
+type ModifiedBy struct {
+	Key string `json:"key,omitempty"`
+	Sig string `json:"sig,omitempty"`
+}
+
+// ResolveOpts holds the options for did resolve.
+type ResolveOpts struct {
+	HTTPClient  *http.Client
+	VersionID   interface{}
+	VersionTime string
+	NoCache     bool
+}
+
+// ResolveOption is a did resolve option.
+type ResolveOption func(opts *ResolveOpts)
+
+// WithHTTPClient the HTTP client input option can be used to resolve with a specific http client.
+func WithHTTPClient(httpClient *http.Client) ResolveOption {
+	return func(opts *ResolveOpts) {
+		opts.HTTPClient = httpClient
+	}
+}
+
+// WithVersionID the version id input option can be used to request a specific version of a DID Document.
+func WithVersionID(versionID interface{}) ResolveOption {
+	return func(opts *ResolveOpts) {
+		opts.VersionID = versionID
+	}
+}
+
+// WithVersionTime the version time input option can used to request a specific version of a DID Document.
+func WithVersionTime(versionTime time.Time) ResolveOption {
+	return func(opts *ResolveOpts) {
+		opts.VersionTime = versionTime.Format(time.RFC3339)
+	}
+}
+
+// WithNoCache the no-cache input option can be used to turn cache on or off.
+func WithNoCache(noCache bool) ResolveOption {
+	return func(opts *ResolveOpts) {
+		opts.NoCache = noCache
+	}
+}
+
+// CreateOpts create did opts.
+type CreateOpts struct {
+	DefaultServiceType     string
+	DefaultServiceEndpoint string
+	EncryptionKey          *did.VerificationMethod
+}
+
+// CreateOption is a create DID option.
+type CreateOption func(opts *CreateOpts)
```

Comment:
How do these Create options intersect with the fact that a DID document is being passed-in?

Reply:
I removed CreateOptions